content/src/controller.js

```diff
@@ -2009,6 +2018,55 @@ function sortByName(a, b) {
 function renderDirectory(position) {
   var pane = paneatpos(position);
   var mpp = model.pane[pane];
+  mpp.bydate = defaultDirSortingByDate();
+  var filename = mpp.filename;
+  view.setPaneLinkText(pane, getUpdatedLinksArray(pane), filename, model.ownername);
+  updateTopControls(false);
+}
+
+
+function updateSortResults(pane) {
+  view.setPaneLinks(pane,getUpdatedLinksArray(pane));
+  updateTopControls(false);
+}
+
+function updateSearchResults(pane, search, cb) {
+  var mpp = model.pane[pane];
+  if(specialowner()) {
+    var cacheResults = cache.get('search-keys', search);
+    if(cacheResults) {
+      mpp.data = cacheResults;
+      view.setPaneLinks(pane, getUpdatedLinksArray(pane));
+      cb && cb();
+    } else {
+      storage.loadFile(model.ownername, mpp.filename+"?prefix="+search, true, function(m) {
```

Comment:
Have you tested what it's like without a cache?

The main problem here is that once we start using prefix search on all directories, it will become impossible to see new files when they are created because the cache is never invalidated.

If we must use a cache, we must find a place to invalidate it.  Maybe after navigating away from a directory listing?